### ISSUE TITLE
Make sure we show a "sold out" message for baskets

### ISSUE THREAD

**Issue by marksteward** [emfcamp/Website#279]
People picking tickets that then sell out get a "you've lost session state" message. They should get a more polite "sorry these tickets are no longer available" message.

**Comment by marksteward** [comment]
This shouldn't happen:

```
2015-12-14_20:20:43.94216 2015-12-14 20:20:43,941 31243 INFO Anon main Created new user with email ... and id: 187
2015-12-14_20:20:43.94834 2015-12-14 20:20:43,948 31243 INFO Anon main Creating tickets for basket [<Ticket None: full>]
2015-12-14_20:20:43.94863 2015-12-14 20:20:43,948 31243 INFO Anon main Payment: GoCardless payment for 100 GBP (ticket total 100)
2015-12-14_20:20:43.94881 2015-12-14 20:20:43,948 31243 INFO Anon main Ticket info: None
2015-12-14_20:20:43.99825 2015-12-14 20:20:43,997 31243 ERROR Anon main Exception on /tickets/pay [POST]
2015-12-14_20:20:43.99828 Traceback (most recent call last):
2015-12-14_20:20:43.99828   File "/var/www/www.emfcamp.org/env/local/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app
2015-12-14_20:20:43.99828     response = self.full_dispatch_request()
2015-12-14_20:20:43.99829   File "/var/www/www.emfcamp.org/env/local/lib/python2.7/site-packages/flask/app.py", line 1477, in full_dispatch_request
2015-12-14_20:20:43.99829     rv = self.handle_user_exception(e)
2015-12-14_20:20:43.99829   File "/var/www/www.emfcamp.org/env/local/lib/python2.7/site-packages/flask/app.py", line 1381, in handle_user_exception
2015-12-14_20:20:43.99830     reraise(exc_type, exc_value, tb)
2015-12-14_20:20:43.99830   File "/var/www/www.emfcamp.org/env/local/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
2015-12-14_20:20:43.99831     rv = self.dispatch_request()
2015-12-14_20:20:43.99831   File "/var/www/www.emfcamp.org/env/local/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
2015-12-14_20:20:43.99831     return self.view_functions[rule.endpoint](**req.view_args)
2015-12-14_20:20:43.99832   File "/var/www/www.emfcamp.org/apps/tickets.py", line 258, in pay
2015-12-14_20:20:43.99832     payment = create_payment(payment_type)
2015-12-14_20:20:43.99833   File "/var/www/www.emfcamp.org/apps/tickets.py", line 68, in create_payment
2015-12-14_20:20:43.99834     db.session.commit()
2015-12-14_20:20:43.99834   File "/var/www/www.emfcamp.org/env/local/lib/python2.7/site-packages/sqlalchemy/orm/scoping.py", line 150, in do
2015-12-14_20:20:43.99834     return getattr(self.registry(), name)(*args, **kwargs)
2015-12-14_20:20:43.99835   File "/var/www/www.emfcamp.org/env/local/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 790, in commit
2015-12-14_20:20:43.99835     self.transaction.commit()
2015-12-14_20:20:43.99836   File "/var/www/www.emfcamp.org/env/local/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 392, in commit
2015-12-14_20:20:43.99836     self._prepare_impl()
2015-12-14_20:20:43.99836   File "/var/www/www.emfcamp.org/env/local/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 372, in _prepare_impl
2015-12-14_20:20:43.99839     self.session.flush()
2015-12-14_20:20:43.99840   File "/var/www/www.emfcamp.org/env/local/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 2004, in flush
2015-12-14_20:20:43.99840     self._flush(objects)
2015-12-14_20:20:43.99840   File "/var/www/www.emfcamp.org/env/local/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 2031, in _flush
2015-12-14_20:20:43.99841     self.dispatch.before_flush(self, flush_context, objects)
2015-12-14_20:20:43.99842   File "/var/www/www.emfcamp.org/env/local/lib/python2.7/site-packages/sqlalchemy/event/attr.py", line 218, in __call__
2015-12-14_20:20:43.99842     fn(*args, **kw)
2015-12-14_20:20:43.99843   File "/var/www/www.emfcamp.org/models/ticket.py", line 382, in check_capacity
2015-12-14_20:20:43.99843     raise TicketError('No more tickets of type %s available' % type.name)
2015-12-14_20:20:43.99844 TicketError: No more tickets of type Full Camp Ticket available
```

**Comment by SamLR** [comment]
I think to solve this we need to move the check_capacity function from where it is. It currently fails by raising an error but this doesn't bubble via anything that can catch it.

A reasonable solution would be to make this check before adding the tickets to the db

**Comment by marksteward** [comment]
Fixed by 447fa32
